src/org/gk/render/AbstractNodeRenderer.java

```diff
@@ -51,32 +52,32 @@ public void render(Graphics g) {
     }
 
     protected abstract void renderShapes(Graphics g);
-    
-    protected void renderDrug(Graphics g) {
+
+    protected void renderDrugLabel(Graphics g) {
         // Want to use a smaller font
         Font font = g.getFont();
         Font oldFond = font;
         font = font.deriveFont(Font.ITALIC + Font.BOLD, font.getSize() - 2);
         Graphics2D g2 = (Graphics2D) g;
         Rectangle2D textBounds = font.getStringBounds(DRUG_SYMBOL, g2.getFontRenderContext());
         // Make sure it is drawn at the bottom right corner
-        Rectangle bounds = node.getBounds();
+        Rectangle labelBounds = node.getBounds();
+
+        // Width and Height of the label.
         int w = (int)(textBounds.getWidth() + boundsBuffer);
         int h = (int)(textBounds.getHeight() + boundsBuffer);
-        int x = (int) (bounds.getMaxX() - w);
-        int y = (int) (bounds.getMaxY() - h);
-        
+        Dimension labelDimensions = new Dimension(w, h);
+
+        // Bottom right coordinates.
+        int x = (int)(labelBounds.getMaxX() - labelDimensions.getWidth());
+        int y = (int)(labelBounds.getMaxY() - labelDimensions.getHeight());
```

Comment:
The above code (lines 66, 67) are much cleaner than what you wrote here. I cannot see any reason why you did this.

Reply:
Fixed in [ad58739](https://github.com/reactome/CuratorTool/pull/26/commits/3822611dfb0486ef3eb45d492b83cc810ad58739).